File: flutter_app/lib/models/medecin.dart
class Medecin {
  final int? id;
  final String nom;
  final String prenom;
  final String specialite;
  final String? numeroOrdre;
  final String? telephone;
  final String? email;
  final int? serviceId;
  final String? serviceNom;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Medecin({
    this.id,
    required this.nom,
    required this.prenom,
    required this.specialite,
    this.numeroOrdre,
    this.telephone,
    this.email,
    this.serviceId,
    this.serviceNom,
    this.createdAt,
    this.updatedAt,
  });

  factory Medecin.fromJson(Map<String, dynamic> json) {
    return Medecin(
      id: json['id'],
      nom: json['nom'] ?? '',
      prenom: json['prenom'] ?? '',
      specialite: json['specialite'] ?? '',
      numeroOrdre: json['numeroOrdre'],
      telephone: json['telephone'],
      email: json['email'],
      serviceId: json['serviceId'],
      serviceNom: json['serviceNom'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'nom': nom,
      'prenom': prenom,
      'specialite': specialite,
      if (numeroOrdre != null) 'numeroOrdre': numeroOrdre,
      if (telephone != null) 'telephone': telephone,
      if (email != null) 'email': email,
      if (serviceId != null) 'serviceId': serviceId,
    };
  }

  String get fullName => '$prenom $nom';
}

File: backend/src/main/java/com/healthcare/dashboard/entities/Medecin.java
package com.healthcare.dashboard.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "medecins")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Medecin {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String nom;
    
    @Column(nullable = false)
    private String prenom;
    
    @Column(unique = true, nullable = false)
    private String numeroInscription;
    
    @Column(nullable = false)
    private String specialite;
    
    @Column
    private String telephone;
    
    @Column
    private String email;
    
    @ManyToOne
    @JoinColumn(name = "service_id")
    private Service service;
    
    @Column
    private String statut = "ACTIF"; // ACTIF, CONGE, INACTIF
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
